cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -186,6 +186,7 @@ if(WIN32)
       #
       # ARROW-2986: Without /EHsc we get C4530 warning
       set(CXX_COMMON_FLAGS "/W3 /EHsc")
+      string(APPEND CMAKE_CXX_FLAGS " /EHsc")
```

Comment:
Why do we need this? `CXX_COMMON_FLAGS` will be appended later: https://github.com/apache/arrow/blob/8afd1403ed147810688f5f50151a6980666fe7e0/cpp/CMakeLists.txt#L526

Reply:
I found this line was needed when I was working on a separate build. Locally on MSVC Visual Studio I didn't need this line, but on CI using Ninja I get C4530 warning when building `GTestAlt`, and that warning was treated as an error. Adding this line resolved the issue below. 

Issue Log:
```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include\__msvc_ostream.hpp(781): error C2220: the following warning is treated as an error
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include\__msvc_ostream.hpp(781): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
```

GitHub Action Run Log: https://github.com/apache/arrow/actions/runs/19112837562/job/54614206545